PubnativeLite/PubnativeLiteDemo/Markup/PNLiteDemoMarkupMainViewController.m

```diff
@@ -0,0 +1,114 @@
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+#import "PNLiteDemoMarkupMainViewController.h"
+#import "PNLiteDemoMarkupDetailViewController.h"
+
+@interface PNLiteDemoMarkupMainViewController ()
+
+@property (weak, nonatomic) IBOutlet UIButton *bannerButton;
+@property (weak, nonatomic) IBOutlet UIButton *mRectButton;
+@property (weak, nonatomic) IBOutlet UIButton *leaderboardButton;
+@property (weak, nonatomic) IBOutlet UIButton *interstitialButton;
+@property (weak, nonatomic) IBOutlet UITextView *markupTextView;
+@property (nonatomic, retain) NSNumber *placement;
+@property (nonatomic, strong) Markup *markup;
+
+@end
+
+@implementation PNLiteDemoMarkupMainViewController
+
+- (void)dealloc {
+    self.markup = nil;
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+}
+
+- (IBAction)loadMarkupTouchUpInside:(UIButton *)sender {
+    if (self.markupTextView.text.length <= 0 || !self.markupTextView.text) {
+        [self showAlertControllerWithMessage:@"Please input some markup."];
+    } else if (!self.placement){
+        [self showAlertControllerWithMessage:@"Please choose a placement."];
+    } else {
+        self.markup = [[Markup alloc] initWithMarkupText:self.markupTextView.text withAdPlacement:self.placement];
+        [self requestAd];
+    }
+}
+
+- (void)requestAd {
+    switch ([self.placement integerValue]) {
+        case 0:
+        case 1:
+        case 2: {
+            PNLiteDemoMarkupDetailViewController *markupDetailVC = [self.storyboard instantiateViewControllerWithIdentifier:@"MarkupDetailViewController"];
+            markupDetailVC.markup = self.markup;
+            [self.navigationController presentViewController:markupDetailVC animated:YES completion:nil];
+            break;
+        }
+        case 3:
+            [self createMRAIDViewWithMarkup:self.markup
+                                  withWidth:[[UIScreen mainScreen] bounds].size.width
```

Comment:
In the end the problem was with the html, right? We can use the screen bounds and not the safe area?

Reply:
I guess so. because normal interstitital I'm using screen bounds and it is working fine. So let's wait for the adtag team to fix it if not, we can change this always 👍